### PR TITLE
STD for code review

### PR DESCRIPTION
**Pull request by ernesto-ca**
Created a new standard for code review general  guidelines, where pro tips and checklists are suggested for the reviewer.

### CODE REVIEW

File: docs/processes/git-code-review.md
Comment (maneroto) [inline docs/processes/git-code-review.md:1]: if this is a standard, then it must go under the standards folder.

File: docs/processes/git-code-review.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+description: Standards about how to review a pull request and/or merge request
+tags: [github, review, code]
+---
+
+# STD - Code Review Guidelines
+
+## Objective
+
+This standard provides guidelines and best practices for reviewing code in a Merge Request (MR) or Pull Request (PR) on GitHub. The objective is to ensure high-quality code, adherence to coding standards, and effective collaboration within the development team.
```
Comment (maneroto) [inline docs/processes/git-code-review.md:10]: ... or Pull Request (PR) in GitHub ...

File: docs/processes/git-code-review.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+## Content
+
+### 1.Introduction
+
+Code reviews play a crucial role in maintaining code quality and fostering collaboration among team members. This document outlines the process and best practices for conducting effective code reviews.
```
Comment (maneroto) [inline docs/processes/git-code-review.md:16]: ... outlines the best practices ...
This has no process structure, but might worth creating one and liniking these standards to the process

File: docs/processes/git-code-review.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+- Existing unit tests are not broken by the proposed changes.
+- Edge cases and error scenarios are covered by tests.
+- If applicable, integration tests and end-to-end tests are included.
+
+#### General Code Standards:
+- The code is consistent with the project's overall architecture.
+- Coding conventions, such as indentation and formatting, are followed.
+- No commented-out code is present in the final submission.
+- Error handling is implemented appropriately.
+- No hardcoded values unless if it require it.
```
Comment (maneroto) [inline docs/processes/git-code-review.md:39]: This point can get ambiguous, you need to specify when hardcoded values are allowed.

Comment (maneroto) [reply docs/processes/git-code-review.md:39]: An option could be to tell "if you think hardcoded values are required, you need to tell your team members to look for other solutions, if no solutions are found, keep hardcoded values, but add a comment to tell that might cause some invonvenients" 
You can also separate the previous into different bullet points or give another approach, but yeah, remove ambiguity

File: docs/processes/git-code-review.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+### 3. Pro Tips for Effective Code Reviews
+
+Offer specific feedback that helps the developer understand the issue and suggests improvements.
```
Comment (maneroto) [inline docs/processes/git-code-review.md:43]: You can also try to explain when to comment a line of code, when to comment a file and when to comment the overall pr/mr

File: docs/processes/git-code-review.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+Maintain a positive and respectful tone in your comments, fostering a collaborative and inclusive environment.
+
+#### II. Consider the Bigger Picture
+Understand the context of the changes and how they fit into the overall project goals.
+
+#### III. Address Security Concerns
+Pay attention to potential security vulnerabilities and address them promptly.
+
+#### IIII. Pair Programming
+Consider occasional pair programming sessions for complex logic or critic vulnerabilities that could be enhanced.
```
Comment (maneroto) [inline docs/processes/git-code-review.md:55]: Explain how does pair programming fits in code reviews

**Review by maneroto — CHANGES_REQUESTED** [review]

File: docs/processes/git-code-review.md (hunk 1 above)
Comment (maneroto) [reply docs/processes/git-code-review.md:10]: Change "on" to "in"

**Review by maneroto — APPROVED** [review]
It still has some spelling errors, but I'll accept it
